extensions/azure-extensions/src/main/java/io/druid/storage/azure/AzureDataSegmentPusher.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.storage.azure;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.metamx.common.CompressionUtils;
+import com.metamx.common.RetryUtils;
+import com.metamx.common.logger.Logger;
+import com.microsoft.azure.storage.StorageException;
+import io.druid.segment.SegmentUtils;
+import io.druid.segment.loading.DataSegmentPusher;
+import io.druid.segment.loading.DataSegmentPusherUtil;
+import io.druid.timeline.DataSegment;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+public class AzureDataSegmentPusher implements DataSegmentPusher
+{
+
+  private static final Logger log = new Logger(AzureDataSegmentPusher.class);
+  public static final String TEMP_DIR = "druid";
+  private final AzureStorageContainer azureStorageContainer;
+  private final AzureAccountConfig azureAccountConfig;
+  private final ObjectMapper jsonMapper;
+
+  @Inject
+  public AzureDataSegmentPusher(
+      AzureStorageContainer azureStorageContainer,
+      AzureAccountConfig azureAccountConfig,
+      ObjectMapper jsonMapper
+  )
+  {
+    this.azureStorageContainer = azureStorageContainer;
+    this.azureAccountConfig = azureAccountConfig;
+    this.jsonMapper = jsonMapper;
+  }
+
+  @Override
+  public String getPathForHadoop(String dataSource)
+  {
+    return null;
+  }
+
+  public File createCompressedSegmentDataFile(final File indexFilesDir) throws IOException
+  {
+    final File zipOutFile = File.createTempFile(TEMP_DIR, AzureStorageDruidModule.INDEX_ZIP_FILE_NAME);
```

Comment:
> are these tempfiles ever cleaned up?

@fjy, I can simplify that using `deleteOnExit`, so the files will be deleted when the Peon's JVM terminates.